connectors/cache/fallback_test.go

```diff
@@ -248,8 +258,17 @@ func TestUpsertCases(t *testing.T) {
 		},
 	}
 	for _, t := range testCases {
-		runTestCase(t)
+		runTestCaseWithInvalidateCache(t)
+	}
+	runTestCaseWithoutInvalidateCache := func(tc testCase) {
```

Comment:
seems like your test should be failing. the test says it's still calling Remove on the fallback

Reply:
there were some issue with the mocking originally, I will help with fixing it.